mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewModel.kt

```diff
@@ -34,402 +18,46 @@ import kotlin.reflect.KProperty1
  * Other classes can observe the state via [stateFlow].
  */
 abstract class MavericksViewModel<S : MavericksState>(
-    initialState: S
+    initialState: S,
+    configFactory: MavericksViewModelConfigFactory = Mavericks.viewModelConfigFactory,
+) : MavericksRepository<S>(
+    initialState,
+    object : MavericksRepositoryConfigProvider {
+        override fun <S : MavericksState> invoke(repository: MavericksRepository<S>, initialState: S): MavericksRepositoryConfig<S> {
+            return configFactory.provideConfig(repository as MavericksViewModel<S>, initialState)
+        }
+    }
 ) {
 
-    // Use the same factory for the life of the viewmodel, as it might change after this viewmodel is created (especially during tests)
-    @PublishedApi
-    internal val configFactory = Mavericks.viewModelConfigFactory
-
-    @Suppress("LeakingThis")
-    @InternalMavericksApi
-    val config: MavericksViewModelConfig<S> = configFactory.provideConfig(
-        this,
-        initialState
-    )
+    val viewModelScope = coroutineScope
```

Comment:
`coroutineScope` also gets inherited by children now, too, right? If so, let's see if we can figure out how to hide that. One idea that would take more code but would work is to have this wrap and delegate to the repo rather than extend it.

Reply:
I thought about using delegate vs inheritance over here, it's for sure will introduce some API duplication but let me check how bad is that

Reply:
Actually I think this is very good idea, as it feels more right that VM has reference to repository vs extends it. Even if it requires some boilerplate glue code this looks more correct.

Reply:
Addressed in https://github.com/airbnb/mavericks/pull/635/commits/7566758ec578ed15bb8ac7534c37f58ef674eff0